src/tools/pkcs11-tool.c

```diff
@@ -2985,9 +2986,6 @@ static int gen_keypair(CK_SLOT_ID slot, CK_SESSION_HANDLE session,
 			size_t mtypes_num = sizeof(mtypes)/sizeof(mtypes[0]);
 			const char *p_param_set = type + strlen("GOSTR3410");
 
-			if (p_param_set == NULL)
-				util_fatal("Unknown key pair type %s, expecting GOSTR3410:<nbytes>", type);
-
```

Comment:
I think you did not want to remove this check as this could lead to using of NULL value in strcmp below and there is no default involved in the gost case.